{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f58583a",
   "metadata": {},
   "outputs": [],
   "source": [
    "n_jo = 8\n",
    "\n",
    "da = \"/Users/kate/craft/data\"\n",
    "\n",
    "to = \"/Users/kate/craft/workflow/Fastq.jl/tool\"\n",
    "\n",
    "be = joinpath(da, \"benchmark\")\n",
    "\n",
    "gr = joinpath(da, \"grch\")\n",
    "\n",
    "vb = joinpath(be, \"HG002_truth/HG002_GRCh38_1_22_v4.2.1_benchmark.vcf.gz\")\n",
    "\n",
    "vc = joinpath(be, \"apply_germline_dna_to_genome/call_germline_variant/pass.vcf.gz\")\n",
    "\n",
    "re = joinpath(gr, \"GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.fna\")\n",
    "\n",
    "sd = replace(re, \"fna\" => \"sdf\")\n",
    "\n",
    "rt = \"/Users/kate/craft/workflow/Fastq.jl/tool/rtg-tools-3.11/rtg\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2ead4cfd",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make vcfeval SDF\n",
    "\n",
    "if !isdir(sd)\n",
    "\n",
    "    run(`$rt format -o $sd $re`)\n",
    "\n",
    "end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "1f2c8560",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Rename chromosomes from numbers to strings\n",
    "\n",
    "vcn = replace(vc, \"pass\" => \"pass_rename_chromosomes\")\n",
    "\n",
    "if !isfile(vcn)\n",
    "\n",
    "    ch = joinpath(gr, \"chromosome/n_chrn.tsv\")\n",
    "\n",
    "    run(`bcftools annotate --threads=$n_jo --rename-chrs=$ch --output=$vcn $vc`)\n",
    "\n",
    "end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "345cc254",
   "metadata": {},
   "outputs": [],
   "source": [
    "## run vcfeval\n",
    "\n",
    "bd = joinpath(be, \"HG002_truth/HG002_GRCh38_1_22_v4.2.1_benchmark_noinconsistent.bed.gz\")\n",
    "\n",
    "ouv = joinpath(be, \"vcfeval\")\n",
    "\n",
    "run(`$rt vcfeval \n",
    "    --baseline=$vb \n",
    "    --bed-regions=$bd \n",
    "    --calls=$vcn \n",
    "    --template=$sd \n",
    "    --output=$ouv\n",
    "    --threads=$n_jo`)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "c98b4a5d",
   "metadata": {},
   "source": [
    "`--baseline` VCF containing baseline truth variants\n",
    "`--calls` VCF containing called variants\n",
    "`--template` SDF of reference genome the the `calls` VCF was made against"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c0652260",
   "metadata": {},
   "outputs": [],
   "source": [
    "## run hap.py in hap.py docker container\n",
    "\n",
    "ouh = joinpath(be, \"happy/\")\n",
    "\n",
    "if !isdir(ouh)\n",
    "\n",
    "    mkdir(ouh)\n",
    "\n",
    "end\n",
    "\n",
    "vto = joinpath(\"/home\", basename(to))\n",
    "\n",
    "vda = joinpath(\"/home\", basename(da))\n",
    "\n",
    "vvb = joinpath(vda, split(vb, \"data/\")[2])\n",
    "\n",
    "vvc = joinpath(vda, split(vcn, \"data/\")[2])\n",
    "\n",
    "vbd = joinpath(vda, split(bd, \"data/\")[2])\n",
    "\n",
    "vouh = joinpath(vda, split(ouh, \"data/\")[2], \"hap.py\")\n",
    "\n",
    "vrt = joinpath(vto, basename(dirname(rt)))\n",
    "\n",
    "vsd = joinpath(vda, split(sd, \"data/\")[2])\n",
    "\n",
    "vre = joinpath(vda, split(re, \"data/\")[2])\n",
    "\n",
    "\n",
    "println(vto)\n",
    "println(vda)\n",
    "println(vvb)\n",
    "println(vvc)\n",
    "println(vbd)\n",
    "println(vouh)\n",
    "println(vrt)\n",
    "println(vsd)\n",
    "println(vre)\n",
    "\n",
    "id = readlines(pipeline(`docker run \n",
    "        --interactive \n",
    "        --detach \n",
    "        --tty \n",
    "        --user root \n",
    "        -v $to:$vto \n",
    "        -v $da:$vda\n",
    "        pkrusche/hap.py\n",
    "        bash`))\n",
    "\n",
    "println(id)\n",
    "\n",
    "readlines(pipeline(`docker exec --interactive $id bash -c \n",
    "                   \"/opt/hap.py/bin/hap.py $vvb $vvc \n",
    "                   -f $vbd \n",
    "                   -r $vre \n",
    "                   -o $vouh \n",
    "                   --engine-vcfeval-path $vrt \n",
    "                   --engine-vcfeval-template $vsd\"`))\n",
    "\n",
    "# this command worked on \n",
    "# ./hap.py /home/data/benchmark/HG002_truth/HG002_GRCh38_1_22_v4.2.1_benchmark.vcf.gz /home/data/benchmark/apply_germline_dna_to_genome/call_germline_variant/pass_rename_chromosomes.vcf.gz -f /home/data/benchmark/HG002_truth/HG002_GRCh38_1_22_v4.2.1_benchmark_noinconsistent.bed.gz -r /home/data/grch/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.fna -o /home/data/benchmark/hap.py --engine-vcfeval-path /home/tool/rtg-tools-3.11 --engine-vcfeval-template /home/data/grch/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.sdf\n",
    "\n",
    "#readlines(\n",
    "#    pipeline(\n",
    "#        `docker exec --interactive $id bash -c \"/opt/hap.py/bin/hap.py $vvb $vvc -f $vbd -r $vre -o $vou -V --engine-vcfeval-path $vrt --engine-vcfeval-template $vsd\"`,\n",
    "#    ),\n",
    "#)\n",
    "\n",
    "# && /opt/hap.py/hap.py $vb $vcn -f $bd -o $vhao -V --verbose --engine-vcfeval-path $vrt --engine-vcfeval-template $sd\""
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Julia 1.7.2",
   "language": "julia",
   "name": "julia-1.7"
  },
  "language_info": {
   "file_extension": ".jl",
   "mimetype": "application/julia",
   "name": "julia",
   "version": "1.7.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
