{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "492ec8c8",
   "metadata": {},
   "source": [
    "Truth dataset HG002 version 3.2.2 or above is best to use for benchmarking.\n",
    "\n",
    "how it was done\n",
    "https://github.com/ga4gh/benchmarking-tools/tree/master/doc/ref-impl"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "9f58583a",
   "metadata": {},
   "outputs": [],
   "source": [
    "da = \"/Users/kate/craft/data/\"\n",
    "\n",
    "re = joinpath(da, \"grch/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.fna.gz\")\n",
    "\n",
    "vb = joinpath(da, \"HG002_truth/HG002_GRCh38_1_22_v4.2.1_benchmark.vcf.gz\")\n",
    "\n",
    "vc = joinpath(da, \"apply_germline_dna_to_genome/call_germline_variant/pass.vcf.gz\")\n",
    "\n",
    "sdf = joinpath(pai, \"GRCh38.sdf\")"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "0e129123",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Make vcfeval SDF\n",
    "rtg = \"/Users/kate/craft/workflow/Fastq.jl/tool/rtg-tools-3.11/rtg\"\n",
    "\n",
    "sd = joinpath(da, \"grch/grch/GCA_000001405.15_GRCh38_no_alt_plus_hs38d1_analysis_set.sdf\")\n",
    "\n",
    "run(`$rtg format -o $sdf $re`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "532f2265",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Configure vcfeval\n",
    "run(`$rtg`)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2fb057c2",
   "metadata": {},
   "outputs": [],
   "source": [
    "# ### Download Real Time Genomics prepard GRCH38 files\n",
    "# vcfeval requires reference files in RTG formats. RTG makes versions of these files for the popular reference genomes such as GRCh38, which will be downloaded below.\n",
    "\n",
    "if ispath(sdf)\n",
    "\n",
    "    println(\"Skipping download becauSE RTG GRCh38 data directory already exists:\\n $sdf\")\n",
    "\n",
    "else\n",
    "\n",
    "    run(\n",
    "        pipeline(\n",
    "            `wget -q -P ../input/ https://s3.amazonaws.com/rtg-dataSEts/references/GRCh38.sdf.zip`,\n",
    "            `unzip -d ../input/ ../input/GRCh38.sdf.zip`,\n",
    "            `rm ../input/GRCh38.sdf.zip`,\n",
    "        ),\n",
    "    )\n",
    "end"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "c6874c00",
   "metadata": {},
   "outputs": [],
   "source": [
    "# Run vcfeval\n",
    "# This command compares the VCF you generated (calls) to the highly confident VCF generated by Genome in a Bottle (baseline).\n",
    "run(`$rtg vcfeval\n",
    "    --baSEline=$vb\n",
    "    --calls=$vc\n",
    "    --template=$sdf\n",
    "    --output=$pab\n",
    "    --threads=1\n",
    "    `)\n",
    "\n",
    "# See results\n",
    "# Read more about how to interpret vcfeval results [here](https://cdn.rawgit.com/RealTimeGenomics/rtg-tools/master/installer/resources/tools/RTGOperationsManual/rtg_command_reference.html#vcfeval).\n",
    "\n",
    "pasu = open(joinpath(pab, \"summary.txt\"))\n",
    "\n",
    "su = read(pasu)\n",
    "\n",
    "su"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Julia 1.7.2",
   "language": "julia",
   "name": "julia-1.7"
  },
  "language_info": {
   "file_extension": ".jl",
   "mimetype": "application/julia",
   "name": "julia",
   "version": "1.7.2"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
