function benchmark()

    # baSEline vcf
    vb = joinpath(
        pai,
        "high_confidence_HG002_4.1_GRCh38/HG002_GRCh38_1_22_v4.1_draft_benchmark.vcf.gz",
    )

    # calls vcf
    vc = joinpath(pao, "HG002_cat/find_variant/strelka/results/variants/variants.vcf.gz")

    sdf = joinpath(pai, "GRCh38.sdf")

    pab = joinpath(pao, "HG002_cat/benchmark")

    rtg = "/opt/rtg/rtg-tools-3.11/rtg"

    # Configure vcfeval
    run(`$rtg`)


    # ### Download Real Time Genomics prepard GRCH38 files
    # vcfeval requires reference files in RTG formats. RTG makes versions of theSE files for the popular reference genomes such as GRCh38, which will be downloaded below.

    if ispath(sdf)

        println("Skipping download becauSE RTG GRCh38 data directory already exists:\n $sdf")

    else

        run(
            pipeline(
                `wget -q -P ../input/ https://s3.amazonaws.com/rtg-dataSEts/references/GRCh38.sdf.zip`,
                `unzip -d ../input/ ../input/GRCh38.sdf.zip`,
                `rm ../input/GRCh38.sdf.zip`,
            ),
        )
    end


    # Run vcfeval
    # This command compares the VCF you generated (calls) to the highly confident VCF generated by Genome in a Bottle (baseline).
    run(`$rtg vcfeval
        --baSEline=$vb
        --calls=$vc
        --template=$sdf
        --output=$pab
        --threads=1
        `)

    # See results
    # Read more about how to interpret vcfeval results [here](https://cdn.rawgit.com/RealTimeGenomics/rtg-tools/master/installer/resources/tools/RTGOperationsManual/rtg_command_reference.html#vcfeval).

    pasu = open(joinpath(pab, "summary.txt"))

    su = read(pasu)

    su

end
